ndcube/ndcube.py

```diff
@@ -548,6 +548,9 @@ def _get_crop_by_values_item(self, *points, units=None, wcs=None):
                     raise ValueError(f"Unit '{unit}' of coordinate object {j} in point {i} is "
                                      f"incompatible with WCS unit '{wcs.world_axis_units[j]}'")
```

Comment:
Lines 545 - 549 can be removed now that the unit will be checked implicitly by the conversion.

Reply:
well I guess this checks they are compatible? what if you passed `km` for example or something else that isnt for example convertible to `u.deg`

Reply:
but I could tidy this up to check before converting

Reply:
The unit conversion would error and tell you that km is not compatible with deg.

If you wanted to try to give a more helpful error message you could do
```python
if value is not None:
    try:
        points[i][j] = points[i][j].to(wcs.world_axis_units[j])
    except UnitsError:
        raise UnitsError(f"Unit '{unit}' of coordinate object {j} in point {i} is "
                         f"incompatible with WCS unit '{wcs.world_axis_units[j]}'"
```
You'd have to import `UnitsError` from `astropy.units`.